src/details/ArborX_DetailsFDBSCAN.hpp

```diff
@@ -49,44 +49,48 @@ struct FDBSCANCallback
   UnionFind _union_find;
   CorePointsType _is_core_point;
 
-  template <typename Query>
-  KOKKOS_FUNCTION auto operator()(Query const &query, int j) const
+  KOKKOS_FUNCTION auto operator()(int i, int j) const
   {
-    int const i = ArborX::getData(query);
-
     bool const is_border_point = !_is_core_point(i);
+    bool const is_neighbor_core_point = _is_core_point(j);
```

Comment:
I see it is old code but I still struggle with that name
Would `neighbor_is_core_point` be acceptable?